### PR TITLE
Feature 18 resources add edit delete UI

### PR DESCRIPTION
**Pull request by dimitrov93**

### CODE REVIEW

File: src/components/AddResources/Gemstone.vue [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  ></v-text-field>
+</template>
+
+<script>
+import { useStore } from "vuex";
+
+export default {
+  setup() {
+    const store = useStore();
+    const formData = store.state.resources.resourceDetails;
```
Comment (VladoKat) [inline src/components/AddResources/Gemstone.vue:73]: I see these are not taken via store getters. Do you know what the difference is and if they should be taken through the getters instead?

File: src/store/modules/resources.js [hunk 2]
```diff
@@ -20,21 +26,68 @@ export default {
...
+    },
+    updateFormData(state, data) {
+      state.resourceDetails = data;
+    },
+    updateResource(state, updatedResource) {
+      const index = state.resources.findIndex(
+        (resource) => resource.id === updatedResource.id
+      );
+      if (index !== -1) {
+        state.resources.splice(index, 1, updatedResource);
```
Comment (VladoKat) [inline src/store/modules/resources.js:57]: Here we are replacing the resource at `index` with `updatedResource`, right? Can't we just do `state.resources[index] = updatedResource`?

File: src/views/Add.vue [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    const selected = ref("");
+    const form = ref(null);
+    const sentData = ref(false);
+    const formData = computed(() => store.state.resources.resourceDetails);
+
+    if (props.id) {
+      const data = computed(() =>store.getters["resources/getResourceById"](props.id));
+      store.dispatch("resources/setResourceDetails", data.value);
+      const resourceDetail = computed(() => store.getters["resources/getResourceDetails"]);
+      selected.value = resourceDetail.value.clazz;
```
Comment (VladoKat) [inline src/views/Add.vue:92]: Let's rename data to resourceDetails. We can also remove the `store.getters["resources/getResourceDetails"])` call as we already have the clazz field from the `store.getters["resources/getResourceById"](props.id)` object. 

The code would become:
```
const resourceDetails = computed(() =>store.getters["resources/getResourceById"](props.id));
store.dispatch("resources/setResourceDetails", resourceDetails.value);
selected.value = resourceDetails.value.clazz;
```

File: src/views/Add.vue [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+      }
+    };
+
+    const handleSubmit = async () => {
+      const { valid } = await form.value.validate();
+      if (valid && pageTitle.value === "Add resource") {
+        store.dispatch("resources/setResourceDetails", {
+          clazz: selected.value,
+          ...formData.value,
+        });
```
Comment (VladoKat) [inline src/views/Add.vue:118]: Do we need to do this store update?

File: src/views/Add.vue [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+          showSnackbar({
+            message: "Failed to send resource.",
+            color: "error",
+            timeout: 4000,
+            location: "top right",
+          });
+        }
+      } else if (valid && pageTitle.value == "Edit resource") {
+        const resourceDetail = computed(
+          () => store.getters["resources/getResourceDetails"]
```
Comment (VladoKat) [inline src/views/Add.vue:136]: This resourceDetail is the same as formData fetched on line 86 (const formData = computed(() => store.state.resources.resourceDetails);)

File: src/views/Add.vue [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    const pageTitle = ref(route.meta.title);
+    const getText = computed(() => (text1, text2) => {
+      return pageTitle.value === "Edit resource" ? text1 : text2;
+    });
+
+    const showSnackbar = inject("showSnackbar");
+    const selected = ref("");
+    const form = ref(null);
+    const sentData = ref(false);
+    const formData = computed(() => store.state.resources.resourceDetails);
```
Comment (VladoKat) [inline src/views/Add.vue:86]: let this be resourceDetails instead of formData

**Review by VladoKat — CHANGES_REQUESTED** [review]
Suggested some changes. Let's discuss them when we have the chance. I might try to fix them on my own if i find some time.

File: src/components/AddResources/Gemstone.vue (hunk 1 above)
Comment (dimitrov93) [reply src/components/AddResources/Gemstone.vue:73]: the only difference is that getters are for more maintainability, organization and if you need to implement some logic.

File: src/store/modules/resources.js (hunk 2 above)
Comment (dimitrov93) [reply src/store/modules/resources.js:57]: yes works

File: src/views/Add.vue [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+          <v-card-text
+            v-if="sentData"
+            class="text-red-darken-4 text-center text-h6 pa-0"
+            style="height: 1rem"
+          >
+            {{
+              getText("Item has been updated", "Information has been submited")
+            }}
+          </v-card-text>
```
[the remark this answers is not in the document]
Comment (dimitrov93) [reply src/views/Add.vue:32]: we can, but as far as Im aware, we will use it to handle errors only?

File: src/views/Add.vue [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+          <div v-else style="height: 1rem"></div>
+
+          <div v-if="selected" class="d-flex flex-column">
+            <v-btn color="success" class="mt-4" block type="submit">
+              {{ getText("Edit", "Submit") }}
```
[the remark this answers is not in the document]
Comment (dimitrov93) [reply src/views/Add.vue:37]: for better user experience, i think it should be like it is now, at least for me its easier to understand whats happening

**Comment by VladoKat** [comment]
Changes are part of https://github.com/JewelleryManagement/jewellery-management-ui/pull/19

**Pull request merged**